File: problem30/p30.java
public class p30 {

	public static int digitFifthPowers(int n, int e){
	    int num, sum = 0, digitSum;

	    for(int i=2; i<n; i++){
	        digitSum = 0;
	        num = i;
	        while(num > 0){
	            digitSum += Math.pow(num % 10, e);
	            num /= 10;
	        }
	        if(digitSum == i){
	            sum += i;
	        }
	    }

	    return sum;
	}
	
	public static void main(String[] args) {
		System.out.println(digitFifthPowers(350000, 5));
	}

}

File: problem30/p30.py
def digit_fifth_powers(limit, n):
    res = 0
    for i in range(2, limit+1):
        num = i
        digit_sum = 0
        while num > 0:
            digit_sum += pow(num % 10, n)
            num //= 10
        if digit_sum == i:
            res += i
    return res


print(digit_fifth_powers(350000, 5))
